src/Compilers/CSharp/Portable/Symbols/Synthesized/GeneratedNames.cs

```diff
@@ -52,9 +52,9 @@ internal static string MakeLambdaDisplayClassName(int methodOrdinal, int generat
             return MakeMethodScopedSynthesizedName(GeneratedNameKind.LambdaDisplayClass, methodOrdinal, generation, suffix: "DisplayClass", entityOrdinal: closureOrdinal, entityGeneration: closureGeneration);
         }
 
-        internal static string MakeAnonymousTypeOrDelegateTemplateName(int index, int submissionSlotIndex, string moduleId, bool isDelegate)
```

Comment:
nit: Since we only have two callers, one with just `generation` and one with all parameters, maybe two overloads would be simpler (no conditional logic)?